state/state.go

```diff
@@ -2168,6 +2168,15 @@ func (st *State) AddRelation(eps ...Endpoint) (r *Relation, err error) {
 			if err != nil {
 				return nil, err
 			}
+			// Enforce max-relation limits for the app:ep combination
+			existingRelationCount, err := enforceMaxRelationLimit(app, ep)
```

Comment:
This isn't ideal, as the "existingRelationCount" is carrying state: `-1`. Therefore if someone else removes the if statement below, it will cause issues if it's not well tested. I wonder if we should either rename the variable, or pass a pool to whether it should enforce the relation count?

```suggestion
			existingRelationCount, shouldEnforce, err := enforceMaxRelationLimit(app, ep)
```
I'm not going to die on this hill though.

Reply:
added a bool to signal that the relation count should be asserted.